src/NUnitFramework/framework/Constraints/EachItemConstraintResult.cs

```diff
@@ -0,0 +1,40 @@
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// Provides a <see cref="ConstraintResult"/> for the constraints 
+    /// that are applied to each item in the collection
+    /// </summary>
+    public class EachItemConstraintResult : ConstraintResult
+    {
+        private readonly object _nonMatchingItem;
+        private readonly int _nonMatchingItemIndex;
+
+        /// <summary>
+        /// Constructs a <see cref="EachItemConstraintResult" /> for a particular <see cref="Constraint" />.
+        /// Only used for Failure
+        /// </summary>
+        /// <param name="constraint">The Constraint to which this result applies.</param>
+        /// <param name="actualValue">The actual value to which the Constraint was applied.</param>
+        /// <param name="nonMatchingItem">Actual item that does not match expected condition</param>
+        /// <param name="nonMatchingIndex">Non matching item index</param>
+        public EachItemConstraintResult(IConstraint constraint, object actualValue, object nonMatchingItem, int nonMatchingIndex)
+            : base(constraint, actualValue, false)
+        {
+            _nonMatchingItem = nonMatchingItem;
+            _nonMatchingItemIndex = nonMatchingIndex;
+        }
+
+        /// <summary>
+        /// Write constraint description, actual items, and non-matching item
+        /// </summary>
+        /// <param name="writer">The MessageWriter on which to display the message</param>
+        public override void WriteMessageTo(MessageWriter writer)
+        {
+            base.WriteMessageTo(writer);
```

Comment:
Is there a reason not to use the `WriteAdditionalLinesTo(MessageWriter writer)` method here, like the other ConstraintResults? e.g. https://github.com/nunit/nunit/blob/ffcfba9bdd85792f6aa32dada77a4b044857b4bf/src/NUnitFramework/framework/Constraints/CollectionEquivalentConstraintResult.cs#L56

Reply:
This method added only a week ago.
Will update accordingly.

Reply:
So it was! I didn't realise! 😅